### DISCUSSION TITLE
github plugin does not use the Personal Access Token

### DISCUSSION THREAD

**Discussion by teodias** [gitextensions/gitextensions#10246] [Q&A]
### Environment

- Git Extensions 3.5.4.12724
- Build 65f01f39982ccb121c4df7d4b00b506939553958
- Git 2.30.2.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET Framework 4.8.4510.0
- DPI 96dpi (no scaling)

### Issue description

I put my Personal Access token created on github into the field in the github plugin.

### Steps to reproduce

pushing to my repo asks EVERYTIME for username and password (attention: github disabled password access).
this way the app very inconvenieant to use.

### Did this work in previous version of GitExtensions?

no

### Diagnostics

-

**Comment by akinwilson** [comment]
**READ IF YOU'RE CREATING A NEW REPOSITORY**

Double check the link you are submitting to be used as an origin.

I was just dealing with this issue and found that I had mistakenly provided the **HTTPS** rather than the **SSH** link as the origin. 

I went back and changed it to be the **SSH** link and it works fine.

**Comment by pmiossec** [comment]
> I put my Personal Access token created on github into the field in the github plugin

The PAT provided in the Github plug-in settings is only used for Github features using github APIs. It is not used for fetch/push operation. 

> pushing to my repo asks EVERYTIME for username and password (attention: github disabled password access). this way the app very inconvenieant to use.

GitExtensions doesn't handle the authentication part. It's git that does it... 

Even if a password is asked, indeed, ow that Github deprecated password access, you need to pass a PAT. 

If the credential are asked everytime, that means you have not installed the Git credential manager during git installation (or you have a problem of configuration). It's not a GitExtensions problem...

**Comment by godcong** [comment]
@teodias hi, your problem is the same as my. #9867

**Comment by teodias** [comment]
as I couldn't find a way to get it working I had to switch smartgit

**Comment by vbjay** [comment]
You have to configure your credential manager in git config. It works just
fine when you set that up.
